2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -117,23 +117,60 @@ function mongo_initiate() {
 
 # get the address of the current primary member
 function mongo_primary_member_addr() {
-  local current_endpoints=$(endpoints)
-  local mongo_node="$(echo "${current_endpoints}" | grep -v "$(container_addr)" | head -1):${CONTAINER_PORT}"
-  echo -n $(mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host ${mongo_node} --quiet --eval "print(rs.isMaster().primary);")
+  endpoints | grep -v "$(container_addr)" | while read mongo_node; do
+    cmd_output="$(mongo admin -u admin -p "$MONGODB_ADMIN_PASSWORD" --host "$mongo_node:$CONTAINER_PORT" --eval 'print(rs.isMaster().primary)' --quiet || true)"
```

Comment:
>Let's try:

If you're OK with swallowing the error message (`Thu Mar  3 13:29:09.090 Error: couldn't connect to server localhost:21071 at src/mongo/shell/mongo.js:147`) then we can use your approach.

>And I really think we don't need the grep part.

In this case you will connect to `Thu` host because today is Thursday. I was trying fix this issue actually :-)